src/__swaps__/screens/Swap/components/TokenList/ChainSelection.tsx

```diff
@@ -30,7 +30,9 @@ export const ChainSelection = memo(function ChainSelection({ allText, output }:
   const { accentColor: accountColor } = useAccountAccentColor();
   const { selectedOutputChainId, setSelectedOutputChainId } = useSwapContext();
 
-  const balanceSortedChainList = userAssetsStore.getState().getBalanceSortedChainList();
+  const { getBalanceSortedChainList, getChainsWithBalance } = userAssetsStore.getState();
+  // chains sorted by balance on output, chains without balance hidden on input
+  const balanceSortedChainList = output ? getBalanceSortedChainList() : getChainsWithBalance();
```

Comment:
Should probably use a selector here right? So that it's reactive to wallet changes.

```suggestion
  // chains sorted by balance on output, chains without balance hidden on input
  const balanceSortedChainList = userAssetsStore(state => (output ? state.getBalanceSortedChainList() : state.getChainsWithBalance()));
```